src/components/mongo.ts

```diff
@@ -0,0 +1,132 @@
+import * as pulumi from '@pulumi/pulumi';
+import * as aws from '@pulumi/aws';
+import * as awsx from '@pulumi/awsx';
+import { Ecs, RoleInlinePolicy } from './ecs-service';
+import { Size } from '../types/size';
+import { commonTags } from '../constants';
+
+export type MongoArgs = {
+  /**
+   * Exposed service port.
+   */
+  port: pulumi.Input<number>;
+  /**
+   * CPU and memory size used for running the container. Defaults to "small".
+   * Available predefined options are:
+   * - small (0.25 vCPU, 0.5 GB memory)
+   * - medium (0.5 vCPU, 1 GB memory)
+   * - large (1 vCPU memory, 2 GB memory)
+   * - xlarge (2 vCPU, 4 GB memory)
+   */
+  size?: pulumi.Input<Size>;
+  /**
+   * The aws.ecs.Cluster resource.
+   */
+  cluster: aws.ecs.Cluster;
+  /**
+   * The awsx.ec2.Vpc resource.
+   */
+  vpc: awsx.ec2.Vpc;
+  /**
```

Comment:
To avoid duplication you can extend MongoArgs with EcsService args. By doing so, you won't need to explain common properties in each service

Reply:
In that case how would we handle port arg? It is not required in mongo service, but it is required in ecs service.

Reply:
Can you use typescript utility types Required, Partial, etc...